app/routes/teams.rb

```diff
@@ -156,6 +156,35 @@ class Teams < Core
         end
       end
 
+      get '/teams/:slug/streams/users/:username/tracks/:id' do |slug, username, track_id|
+        please_login
+        only_with_existing_team(slug) do |team|
+          unless team.includes?(current_user)
+            flash[:error] = "You may only view team pages for teams that you are a member of, or that you manage."
+            redirect '/'
+          end
+
+          user = ::User.find_by_username(username)
```

Comment:
Nowadays it's recommended to use `User.find_by(username: username)` instead of dynamic finders. :)